### PR TITLE
SVAR Doc

### PR DESCRIPTION
**Pull request by ajain-work**

### CODE REVIEW

File: doc/source/api/solver/svar.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
+.. _ref_svar_data:
+
+SVAR data
```
Comment (ajain-work) [inline doc/source/api/solver/svar.rst:3]: @PipKat Kindly review.

File: doc/source/api/solver/svar.rst [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  >>>svar_info_centroid.dimension 
+  >>>3
+  >>>
+  >>>svar_info_centroid.field_type 
+  >>><class 'numpy.float64'>  
+  
+SVAR data
+---------
+SVAR Data can be extracted and modified with svar_data object.
+Here are the methods for interacting with SVARs Data:
```
Comment (ajain-work) [inline doc/source/api/solver/svar.rst:99]:
```suggestion
Here are the methods for interacting with SVAR Data:
```

File: doc/source/api/solver/svar.rst [hunk 3]
```diff
@@ -0,0 +1,160 @@
+.. _ref_svar_data:
+
+SVAR data
+=========
+
+You can use SVAR objects that is svar_info and svar_data to access 
+Fluent SVARs info and SVAR data.
```
[the remark this answers is not in the document]
Comment (seanpearsonuk) [reply doc/source/api/solver/svar.rst]: I think we should start by saying briefly what SVAR is before saying that you can access them. I looked at the online Fluent docs to find a concise definition of SVAR to precede the above sentence, but it was very "technical" and doesn't readily carry across to PyFluent documentation. We might insert something like "An SVAR is an array variable that stores the solved data for a particular field".

File: doc/source/api/solver/svar.rst [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  >>> zone_info.count 
+  >>> 3630 
+  >>>
+  >>> zone_info.zone_id 
+  >>> 3
+  >>>
+  >>> zone_info.zone_type 
+  >>> wall  
+  
+Get SVARs info
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:70]: Maybe singular is better?

```suggestion
Get SVAR info
```

File: doc/source/api/solver/svar.rst [hunk 5]
```diff
@@ -0,0 +1,160 @@
+.. _ref_svar_data:
+
+SVAR data
+=========
+
+You can use svar_info and svar_data objects to access Fluent SVAR
+info and data respectively.
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:7]:
```suggestion
An SVAR is an array variable that holds the data for a particular
solved field (such as pressure or velocity). You can use svar_info
and svar_data objects to access Fluent SVAR info and data respectively.
```

File: doc/source/api/solver/svar.rst [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+---------
+SVARs Info can be extracted with svar_info object.
+Here are the methods for requesting SVARs Info:
+
+- ``get_zones_info`` for zones info.
+- ``get_svars_info`` for SVARs info.
+
+Get zones info
+~~~~~~~~~~~~~~~
+You can request zones info by calling the ``get_zones_info`` method.
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:43]:
```suggestion
Get zone information
~~~~~~~~~~~~~~~
You can access zone information by calling the ``get_zones_info`` method.
```

File: doc/source/api/solver/svar.rst [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  >>> zone_info.zone_id 
+  >>> 3
+  >>>
+  >>> zone_info.zone_type 
+  >>> wall  
+  
+Get SVAR info
+~~~~~~~~~~~~~~~~
+You can request SVARs info for a given ``domain_name`` and list of ``zone_names``
+and by calling the ``get_svars_info`` method.
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:74]:
```suggestion
Get SVAR information
~~~~~~~~~~~~~~~~
You can request SVARs information for a given ``domain_name`` and list of ``zone_names``
by calling the ``get_svars_info`` method.
```

File: doc/source/api/solver/svar.rst [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+  >>>svar_info_centroid.dimension 
+  >>>3
+  >>>
+  >>>svar_info_centroid.field_type 
+  >>><class 'numpy.float64'>  
+  
+SVAR data
+---------
+SVAR Data can be extracted and modified with svar_data object.
+Here are the methods for interacting with SVAR Data:
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:100]:
```suggestion
SVAR data
---------
SVAR data can be extracted and modified via the following svar_data object methods:
```

File: doc/source/api/solver/svar.rst [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+
+- ``get_svar_data`` to get SVAR data.
+- ``set_svar_data`` to set SVAR data.
+
+
+Get SVAR data
+~~~~~~~~~~~~~
+You can request SVAR data for a given ``domain_name`` and multiple ``zone_names`` by calling
+the ``get_svar_data`` method and passing required ``svar_name``.
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:109]:
```suggestion
the ``get_svar_data`` method and passing the particular ``svar_name``.
```

File: doc/source/api/solver/svar.rst [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+Additionally svar_data object also supports ``get_array`` method. This method can be used to 
+generate ``numpy zeros array`` for a given ``domain_name``, ``zone_name`` and 
+``svar_name``. This array can be populated and passed to ``set_svar_data``.
+
+.. code-block:: python
+  
+    >>> wall_temp_array = svar_data.get_array("SV_T", "wall", "mixture")
+    >>> fluid_temp_array =svar_data.get_array("SV_T", "fluid", "mixture")
+    >>> wall_temp_array[:]= 500
+    >>> fluid_temp_array[:]= 600
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:147]:
```suggestion
    >>> wall_temp_array = svar_data.get_array("SV_T", "wall", "mixture")
    >>> fluid_temp_array = svar_data.get_array("SV_T", "fluid", "mixture")
    >>> wall_temp_array[:] = 500
    >>> fluid_temp_array[:] = 600
```

File: doc/source/api/solver/svar.rst [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+  >>> wall   
+  >>>
+  >>> zone_info.count 
+  >>> 3630 
+  >>>
+  >>> zone_info.zone_id 
+  >>> 3
+  >>>
+  >>> zone_info.zone_type 
+  >>> wall
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:69]:
```suggestion
  >>> zones_info = svar_info.get_zones_info()
  >>> zones_info.domains
  ['mixture']  
  >>>
  >>> zones_info.zones
  ['fluid', 'wall', 'symmetry', 'pressure-outlet-7', 'velocity-inlet-6', 'velocity-inlet-5', 'default-interior']
  >>>
  >>> zone_info = zones_info['wall']
  >>>
  >>> zone_info
  name:wall count: 3630 zone_id:3 zone_type:wall thread_type:Face
  >>>
  >>> zone_info.name 
  'wall'   
  >>>
  >>> zone_info.count 
  3630 
  >>>
  >>> zone_info.zone_id 
  3
  >>>
  >>> zone_info.zone_type 
  'wall'  
```

File: doc/source/api/solver/svar.rst [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+  >>> name:SV_CENTROID dimension:3 field_type:<class 'numpy.float64'>
+  >>>
+  >>>svar_info_centroid.name 
+  >>>SV_CENTROID
+  >>>
+  >>>svar_info_centroid.dimension 
+  >>>3
+  >>>
+  >>>svar_info_centroid.field_type 
+  >>><class 'numpy.float64'>
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:95]:
```suggestion
  >>> svars_info_wall_fluid = svar_info.get_svars_info(zone_names=['wall' , "fluid"], domain_name="mixture")
  >>>
  >>> svars_info_wall_fluid.svars
  ['SV_CENTROID', 'SV_D', 'SV_H', 'SV_K', 'SV_P', 'SV_T', 'SV_U', 'SV_V', 'SV_W']
  >>>
  >>> svar_info_centroid = svars_info_wall_fluid['SV_CENTROID']
  >>>
  >>> svar_info_centroid
  name:SV_CENTROID dimension:3 field_type:<class 'numpy.float64'>
  >>>
  >>>svar_info_centroid.name 
  'SV_CENTROID'
  >>>
  >>>svar_info_centroid.dimension 
  >>>3
  >>>
  >>>svar_info_centroid.field_type 
  <class 'numpy.float64'> 
```

File: doc/source/api/solver/svar.rst [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+    >>> fluid_temp = sv_t_wall_fluid['fluid']
+    >>>
+    >>> fluid_temp.size
+    >>> 13852
+    >>>
+    >>> fluid_temp.dtype
+    >>> float64
+    >>>
+    >>> fluid_temp
+    >>> array([600., 600., 600., ..., 600., 600., 600.])
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:130]:
```suggestion
    >>> sv_t_wall_fluid= svar_data.get_svar_data(svar_name="SV_T", zone_names=["fluid", "wall"], domain_name="mixture")
    >>>
    >>> sv_t_wall_fluid.domain
    'mixture'
    >>>
    >>> sv_t_wall_fluid.zones
    ['fluid', 'wall']
    >>>
    >>> fluid_temp = sv_t_wall_fluid['fluid']
    >>>
    >>> fluid_temp.size
    13852
    >>>
    >>> fluid_temp.dtype
    'float64'
    >>>
    >>> fluid_temp
    array([600., 600., 600., ..., 600., 600., 600.])
```

File: doc/source/api/solver/svar.rst [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+
+.. autosummary::
+    :toctree: _autosummary
+    :template: flobject-class-template.rst
+    :recursive:
+
+    svar_info.get_zones_info
+    svar_info.get_svars_info
+    svar_data.get_svar_data
+    svar_data.set_svar_data
```
Comment (seanpearsonuk) [inline doc/source/api/solver/svar.rst:160]: Is there something like a table we can copy from somewhere showing what each SV_<*> stands for?

Comment (ajain-work) [reply doc/source/api/solver/svar.rst]: I am not sure but  check if table can be generated.

**Review by prmukherj — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

File: doc/source/api/solver/svar.rst [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+
+SVAR info
+---------
+SVAR metadata information can be accessed via the following svar_info methods:
+
+- ``get_zones_info`` for zone information.
+- ``get_svars_info`` for SVAR information.
+
+Get zone information
+~~~~~~~~~~~~~~~
```
Comment (dnwillia-work) [inline doc/source/api/solver/svar.rst:41]: Missing some `~` chars

File: doc/source/api/solver/svar.rst [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+  3630 
+  >>>
+  >>> zone_info.zone_id 
+  3
+  >>>
+  >>> zone_info.zone_type 
+  'wall'  
+  
+Get SVAR information
+~~~~~~~~~~~~~~~~
```
Comment (dnwillia-work) [inline doc/source/api/solver/svar.rst:71]: Missing `~`

File: doc/source/api/solver/svar.rst [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+- ``get_svars_info`` for SVAR information.
+
+Get zone information
+~~~~~~~~~~~~~~~
+You can access zone information by calling the ``get_zones_info`` method.
+
+.. code-block:: python
+  
+  >>> zones_info = svar_info.get_zones_info()
+  >>> zones_info.domains
```
Comment (dnwillia-work) [inline doc/source/api/solver/svar.rst:47]: Are `domains` documented anywhere?

**Pull request merged**